zebra-chain/src/parameters/network/subsidy.rs

```diff
@@ -391,8 +408,38 @@ impl ParameterSubsidy for Network {
             Network::Mainnet => NetworkUpgrade::Canopy
                 .activation_height(self)
                 .expect("canopy activation height should be available"),
-            // TODO: Check what zcashd does here, consider adding a field to `testnet::Parameters` to make this configurable.
-            Network::Testnet(_params) => FIRST_HALVING_TESTNET,
+            Network::Testnet(params) => {
+                if params.is_regtest() {
+                    FIRST_HALVING_REGTEST
+                } else if params.is_default_testnet() {
+                    FIRST_HALVING_TESTNET
+                } else {
+                    Height(
+                        self.post_blossom_halving_interval()
+                            .try_into()
+                            .expect("post blossom halving interval should fit in a Height"),
+                    )
```

Comment:
This part's not right, but neither is returning `FIRST_HALVING_TESTNET` as it's doing right now.

We could leave it as it is (returning `FIRST_HALVING_TESTNET`) or add it as a field on `testnet::Parameters`.